### PR TITLE
Create ``start``, ``stop``, and ``collect`` routes between Dashboard and ``Manager`` HTTP. Closes #94

### LINKED ISSUE #94: Create `start`, `stop`, and `collect` routes between Dashboard and Manager HTTP

### PR DESCRIPTION
**Pull request by edavalosanaya**
BEWARE: This PR is dependent on PR #115, meaning that this previous PR needs to be merged first before this one.

### CODE REVIEW

**Comment by edavalosanaya** [comment]
Rebased branch to `origin/main`.

File: chimerapy/api.py [hunk 1]
```diff
@@ -16,11 +31,67 @@ def __init__(self, manager: Manager):
...
+            self.manager.state.running = True
+            return web.HTTPOk()
+        else:
+            return web.HTTPError()
+
+    async def post_collect(self, request: web.Request):
+
+        # First, request Nodes to save their data
+        if self.manager.state.collecting:
+            return web.HTTPBadRequest()
```
Comment (umesh-timalsina) [inline chimerapy/api.py:74]: Is this really a bad request?

File: chimerapy/api.py [hunk 2]
```diff
@@ -16,11 +31,67 @@ def __init__(self, manager: Manager):
...
+            self.manager.state.collecting = True
+
+            success = await self.manager.async_broadcast_request(
+                htype="post", route="/nodes/save"
+            )
+            logger.debug(success)
+
+            if success:
+                task = asyncio.create_task(self.manager.async_collect())
+                self.tasks.append(task)
```
Comment (umesh-timalsina) [inline chimerapy/api.py:85]: What happens to these tasks eventually?

File: chimerapy/manager.py [hunk 3]
```diff
@@ -446,6 +449,71 @@ def request_start(url):
...
+        success = await self.async_broadcast_request(
+            htype="post",
+            route="/nodes/collect",
+            data={"path": str(self.logdir)},
+        )
+        self.state.collecting = False
+
+        if success:
+            self.server.move_transfer_files(self.logdir, True)
+            self.save_meta()
```
Comment (umesh-timalsina) [inline chimerapy/manager.py:504]: Will a try-catch block be necessary here?

File: chimerapy/api.py [hunk 4]
```diff
@@ -16,11 +31,67 @@ def __init__(self, manager: Manager):
...
+    async def post_start(self, request: web.Request):
+
+        # Mark the start time
+        self.manager.start_time = datetime.datetime.now()
+
+        # Request start from all Workers
+        success = await self.manager.async_broadcast_request(
+            htype="post", route="/nodes/start"
+        )
+        logger.debug(success)
```
[the remark this answers is not in the document]
Comment (edavalosanaya) [reply chimerapy/api.py:62]: `self.manager.start` would be a synchronous-only routine. The implementation inside `api.py` is completely asynchronous.

File: chimerapy/api.py (hunk 1 above)
Comment (edavalosanaya) [reply chimerapy/api.py:74]: Good point

File: chimerapy/manager.py (hunk 3 above)
Comment (edavalosanaya) [reply chimerapy/manager.py:504]: Good points, a possible point of error

File: chimerapy/api.py (hunk 2 above)
Comment (edavalosanaya) [reply chimerapy/api.py:85]: Do we want to add a shutdown routine that waits until these tasks are completed?

File: chimerapy/api.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (umesh-timalsina) [reply chimerapy/api.py:62]: I see, I was hoping that we could use something like the task created in https://github.com/oele-isis-vanderbilt/ChimeraPy/pull/117/files#diff-34e5b7e05e37275a89fb71ec2aabe286b31a6261c2778d1fb4762ee5be2f9585R83-R85.

[the remark this answers is not in the document]
Comment (umesh-timalsina) [reply chimerapy/api.py:62]: Or, something like https://github.com/oele-isis-vanderbilt/ChimeraPy/blob/fee1efea8eaeb224e971eae6afd3c1c7f250ac7b/chimerapy/networking/async_loop_thread.py#L48C22-L58.

[the remark this answers is not in the document]
Comment (edavalosanaya) [reply chimerapy/api.py:76]: Agreed, I added in my latest commit a way to submit coroutines to the `ThreadedAsyncLoop` and retrieve a `Future` that can be synchronously waited.

**Review by umesh-timalsina — APPROVED** [review]
The state changes and the Ok responses can be dealt with later on in #146. LGTM!!

**Pull request merged**